// aws-amplify-angular/dist/src/components/storage/s3-album-component/s3-album.component.core.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-album\">\n  <div class=\"amplify-album-container\">\n    <amplify-s3-image-core\n      class=\"amplify-image-container\"\n      *ngFor=\"let item of list\"\n      path=\"{{item.path}}\"\n      (selected)=\"onImageSelected($event)\"\n    ></amplify-s3-image-core>\n  </div>\n</div>\n";
var S3AlbumComponentCore = /** @class */ (function () {
    function S3AlbumComponentCore(amplifyService) {
        this.selected = new EventEmitter();
        this.amplifyService = amplifyService;
    }
    S3AlbumComponentCore.prototype.onImageSelected = function (event) {
        this.selected.emit(event);
    };
    Object.defineProperty(S3AlbumComponentCore.prototype, "data", {
        set: function (data) {
            if (!data.path) {
                return;
            }
            var that = this;
            this.amplifyService.storage()
                .list(data.path)
                .then(function (res) {
                that.list = res.map(function (item) {
                    return { path: item.key };
                });
            })
                .catch(function (err) { return console.error(err); });
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(S3AlbumComponentCore.prototype, "path", {
        set: function (path) {
            if (!path) {
                return;
            }
            var that = this;
            this.amplifyService.storage()
                .list(path)
                .then(function (data) {
                that.list = data.map(function (item) {
                    return { path: item.key };
                });
            })
                .catch(function (err) { return console.error(err); });
        },
        enumerable: true,
        configurable: true
    });
    S3AlbumComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-s3-album-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    S3AlbumComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    S3AlbumComponentCore.propDecorators = {
        "selected": [{ type: Output },],
        "data": [{ type: Input },],
        "path": [{ type: Input },],
    };
    return S3AlbumComponentCore;
}());
export { S3AlbumComponentCore };
