// aws-amplify-angular/dist/src/components/interactions/chatbot/chatbot.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver, Output, EventEmitter } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { ChatbotComponentIonic } from './chatbot.component.ionic';
import { ChatbotComponentCore } from './chatbot.component.core';
var ChatBotComponent = /** @class */ (function () {
    function ChatBotComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.complete = new EventEmitter();
    }
    ChatBotComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    ChatBotComponent.prototype.ngOnDestroy = function () { };
    ChatBotComponent.prototype.loadComponent = function () {
        var _this = this;
        var interactionParams = {
            bot: this.bot,
            title: this.title,
            clearComplete: this.clearComplete
        };
        var interactionComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(ChatbotComponentIonic, interactionParams) : new ComponentMount(ChatbotComponentCore, interactionParams);
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(interactionComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = interactionComponent.data;
        componentRef.instance.complete.subscribe(function (e) {
            _this.complete.emit(e);
        });
    };
    ChatBotComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-interactions',
                    template: "\n              <div class=\"amplify-component\">\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    ChatBotComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    ChatBotComponent.propDecorators = {
        "framework": [{ type: Input },],
        "bot": [{ type: Input },],
        "title": [{ type: Input },],
        "clearComplete": [{ type: Input },],
        "complete": [{ type: Output },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return ChatBotComponent;
}());
export { ChatBotComponent };
