// aws-amplify-angular/dist/src/components/authenticator/greeting-component/greeting.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { GreetingComponentIonic } from './greeting.component.ionic';
import { GreetingComponentCore } from './greeting.component.core';
var GreetingComponent = /** @class */ (function () {
    function GreetingComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    GreetingComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    GreetingComponent.prototype.ngOnDestroy = function () { };
    GreetingComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(GreetingComponentIonic, { authState: this.authState }) : new ComponentMount(GreetingComponentCore, { authState: this.authState });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    GreetingComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-greetings',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    GreetingComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    GreetingComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return GreetingComponent;
}());
export { GreetingComponent };
