// aws-amplify-angular/dist/src/components/storage/photo-picker-component/photo-picker.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver, Output, EventEmitter } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { PhotoPickerIonicComponent } from './photo-picker.component.ionic';
import { PhotoPickerComponentCore } from './photo-picker.component.core';
var PhotoPickerComponent = /** @class */ (function () {
    function PhotoPickerComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.picked = new EventEmitter();
        this.loaded = new EventEmitter();
        this.uploaded = new EventEmitter();
    }
    PhotoPickerComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    PhotoPickerComponent.prototype.ngOnDestroy = function () { };
    PhotoPickerComponent.prototype.loadComponent = function () {
        var _this = this;
        var photoPickerComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(PhotoPickerIonicComponent, { url: this.url, path: this.path }) : new ComponentMount(PhotoPickerComponentCore, { url: this.url, path: this.path });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(photoPickerComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = photoPickerComponent.data;
        componentRef.instance.picked.subscribe(function (e) {
            _this.picked.emit(e);
        });
        componentRef.instance.loaded.subscribe(function (e) {
            _this.loaded.emit(e);
        });
        componentRef.instance.uploaded.subscribe(function (e) {
            _this.uploaded.emit(e);
        });
    };
    PhotoPickerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-photo-picker',
                    template: "\n              <div class='amplify-component'>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    PhotoPickerComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    PhotoPickerComponent.propDecorators = {
        "framework": [{ type: Input },],
        "url": [{ type: Input },],
        "path": [{ type: Input },],
        "picked": [{ type: Output },],
        "loaded": [{ type: Output },],
        "uploaded": [{ type: Output },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return PhotoPickerComponent;
}());
export { PhotoPickerComponent };
