// aws-amplify-angular/dist/src/components/authenticator/require-new-password-component/require-new-password.component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-container\" *ngIf=\"_show\">\n<div class=\"amplify-form-container\">\n  <div class=\"amplify-form-body\">\n  <div class=\"amplify-form-header\">You are required to update your password</div>\n    <div class=\"amplify-form-row\">\n      <label class=\"amplify-input-label\" for=\"password\"> Password *</label>\n      <input #password\n        (keyup)=\"setPassword(password.value)\"\n        (keyup.enter)=\"onSubmit()\"\n        class=\"amplify-form-input\"\n        type=\"password\"\n        placeholder=\"Password\"\n      />\n    </div>\n    <div class=\"amplify-form-actions\">\n      <div class=\"amplify-form-cell-left\">\n        <a class=\"amplify-form-link\"\n          (click)=\"onSignIn()\"\n        >Back to Sign In</a>\n      </div>\n      <div class=\"amplify-form-cell-right\">\n        <button class=\"amplify-form-button\"\n          (click)=\"onSubmit()\"\n        >Submit</button>\n      </div>\n    </div>\n  </div>\n  \n</div>\n<div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n<div class=\"amplify-alert-body\">\n  <span class=\"amplify-alert-icon\">&#9888;</span>\n  <div class=\"amplify-alert-message\">{{ errorMessage }}</div>\n  <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n</div>\n</div>\n</div>\n";
var RequireNewPasswordComponentCore = /** @class */ (function () {
    function RequireNewPasswordComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(RequireNewPasswordComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'requireNewPassword';
        },
        enumerable: true,
        configurable: true
    });
    RequireNewPasswordComponentCore.prototype.setPassword = function (password) {
        this.password = password;
    };
    RequireNewPasswordComponentCore.prototype.onSubmit = function () {
        var _this = this;
        var user = this._authState.user;
        var requiredAttributes = user.challengeParam.requiredAttributes;
        this.amplifyService.auth()
            .completeNewPassword(user, this.password, requiredAttributes)
            .then(function () {
            _this.amplifyService.setAuthState({ state: 'signIn', user: user });
        })
            .catch(function (err) { return _this._setError(err); });
    };
    RequireNewPasswordComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    RequireNewPasswordComponentCore.prototype.onAlertClose = function () {
        this._setError(null);
    };
    RequireNewPasswordComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    RequireNewPasswordComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-require-new-password-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    RequireNewPasswordComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    RequireNewPasswordComponentCore.propDecorators = {
        "authState": [{ type: Input },],
    };
    return RequireNewPasswordComponentCore;
}());
export { RequireNewPasswordComponentCore };
