// aws-amplify-angular/dist/src/components/authenticator/require-new-password-component/require-new-password.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { RequireNewPasswordComponentIonic } from './require-new-password.component.ionic';
import { RequireNewPasswordComponentCore } from './require-new-password.component.core';
var RequireNewPasswordComponent = /** @class */ (function () {
    function RequireNewPasswordComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    RequireNewPasswordComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    RequireNewPasswordComponent.prototype.ngOnDestroy = function () { };
    RequireNewPasswordComponent.prototype.loadComponent = function () {
        var requireNewPasswordComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(RequireNewPasswordComponentIonic, { authState: this.authState }) : new ComponentMount(RequireNewPasswordComponentCore, { authState: this.authState });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(requireNewPasswordComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = requireNewPasswordComponent.data;
    };
    RequireNewPasswordComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-require-new-password',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    RequireNewPasswordComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    RequireNewPasswordComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return RequireNewPasswordComponent;
}());
export { RequireNewPasswordComponent };
