// aws-amplify-angular/dist/src/components/authenticator/confirm-sign-in-component/confirm-sign-in.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { ConfirmSignInComponentIonic } from './confirm-sign-in-component.ionic';
import { ConfirmSignInComponentCore } from './confirm-sign-in-component.core';
var ConfirmSignInComponent = /** @class */ (function () {
    function ConfirmSignInComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    ConfirmSignInComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    ConfirmSignInComponent.prototype.ngOnDestroy = function () { };
    ConfirmSignInComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(ConfirmSignInComponentIonic, { authState: this.authState }) : new ComponentMount(ConfirmSignInComponentCore, { authState: this.authState });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    ConfirmSignInComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-confirm-sign-in',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    ConfirmSignInComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    ConfirmSignInComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return ConfirmSignInComponent;
}());
export { ConfirmSignInComponent };
