// aws-amplify-angular/dist/src/components/authenticator/confirm-sign-up-component/confirm-sign-up.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { ConfirmSignUpComponentIonic } from './confirm-sign-up.component.ionic';
import { ConfirmSignUpComponentCore } from './confirm-sign-up.component.core';
var ConfirmSignUpComponent = /** @class */ (function () {
    function ConfirmSignUpComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    ConfirmSignUpComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    ConfirmSignUpComponent.prototype.ngOnDestroy = function () { };
    ConfirmSignUpComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(ConfirmSignUpComponentIonic, { authState: this.authState }) : new ComponentMount(ConfirmSignUpComponentCore, { authState: this.authState });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    ConfirmSignUpComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-confirm-sign-up',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    ConfirmSignUpComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    ConfirmSignUpComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return ConfirmSignUpComponent;
}());
export { ConfirmSignUpComponent };
