// aws-amplify-angular/dist/src/components/authenticator/authenticator/authenticator.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { AuthenticatorIonicComponent } from './authenticator.component.ionic';
import { AuthenticatorComponentCore } from './authenticator.component.core';
var AuthenticatorComponent = /** @class */ (function () {
    function AuthenticatorComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.hide = [];
    }
    AuthenticatorComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    AuthenticatorComponent.prototype.ngOnDestroy = function () { };
    AuthenticatorComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(AuthenticatorIonicComponent, { hide: this.hide }) : new ComponentMount(AuthenticatorComponentCore, { hide: this.hide });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    AuthenticatorComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-authenticator',
                    template: "\n              <div class=\"amplify-component\">\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    AuthenticatorComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    AuthenticatorComponent.propDecorators = {
        "framework": [{ type: Input },],
        "hide": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return AuthenticatorComponent;
}());
export { AuthenticatorComponent };
