// aws-amplify-angular/dist/src/components/authenticator/authenticator/authenticator.component.core.js
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n\n  <div class=\"amplify-authenticator\">\n\n    <amplify-auth-sign-in-core\n      *ngIf=\"!shouldHide('SignIn')\"\n      [authState]=\"authState\"\n    ></amplify-auth-sign-in-core>\n\n    <amplify-auth-sign-up-core\n      *ngIf=\"!shouldHide('SignUp')\"\n      [authState]=\"authState\"\n    ></amplify-auth-sign-up-core>\n\n    <amplify-auth-confirm-sign-up-core\n      *ngIf=\"!shouldHide('ConfirmSignUp')\"\n      [authState]=\"authState\"\n    ></amplify-auth-confirm-sign-up-core>\n\n    <amplify-auth-confirm-sign-in-core\n    *ngIf=\"!shouldHide('ConfirmSignIn')\"\n    [authState]=\"authState\"\n    ></amplify-auth-confirm-sign-in-core>\n\n    <amplify-auth-forgot-password-core\n    *ngIf=\"!shouldHide('ForgotPassword')\"\n    [authState]=\"authState\"\n    ></amplify-auth-forgot-password-core>\n\n    <amplify-auth-greetings-core\n    *ngIf=\"!shouldHide('Greetings')\"\n    [authState]=\"authState\"\n    ></amplify-auth-greetings-core>\n\n     <amplify-auth-require-new-password-core\n    *ngIf=\"!shouldHide('RequireNewPassword')\"\n    [authState]=\"authState\"\n    ></amplify-auth-require-new-password-core>\n  </div>\n";
var AuthenticatorComponentCore = /** @class */ (function () {
    function AuthenticatorComponentCore(amplifyService) {
        this.authState = {
            state: 'signIn',
            user: null
        };
        this.hide = [];
        this.amplifyService = amplifyService;
        this.subscribe();
    }
    AuthenticatorComponentCore.prototype.subscribe = function () {
        var _this = this;
        this.amplifyService.authStateChange$
            .subscribe(function (state) {
            _this.authState = state;
        }, function () {
            _this.authState = {
                'state': 'signIn',
                'user': null
            };
        });
    };
    AuthenticatorComponentCore.prototype.shouldHide = function (comp) {
        return this.hide.filter(function (item) { return item === comp; })
            .length > 0;
    };
    AuthenticatorComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-authenticator-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    AuthenticatorComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    AuthenticatorComponentCore.propDecorators = {
        "hide": [{ type: Input },],
    };
    return AuthenticatorComponentCore;
}());
export { AuthenticatorComponentCore };
