// aws-amplify-angular/dist/src/components/authenticator/sign-up-component/sign-up.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { SignUpComponentIonic } from './sign-up.component.ionic';
import { SignUpComponentCore } from './sign-up.component.core';
var SignUpComponent = /** @class */ (function () {
    function SignUpComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    SignUpComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    SignUpComponent.prototype.ngOnDestroy = function () { };
    SignUpComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(SignUpComponentIonic, { authState: this.authState }) : new ComponentMount(SignUpComponentCore, { authState: this.authState });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    SignUpComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-sign-up',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    SignUpComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    SignUpComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return SignUpComponent;
}());
export { SignUpComponent };
