// aws-amplify-angular/dist/src/components/authenticator/sign-in-component/sign-in.component.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { SignInComponentIonic } from './sign-in.component.ionic';
import { SignInComponentCore } from './sign-in.component.core';
var SignInComponent = /** @class */ (function () {
    function SignInComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    SignInComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    SignInComponent.prototype.ngOnDestroy = function () { };
    SignInComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework === 'ionic' ? new ComponentMount(SignInComponentIonic, { authState: this.authState }) : new ComponentMount(SignInComponentCore, { authState: this.authState });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    SignInComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-sign-in',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    SignInComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    SignInComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return SignInComponent;
}());
export { SignInComponent };
