// aws-amplify-angular/dist/src/components/authenticator/forgot-password-component/forgot-password.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { ForgotPasswordComponentIonic } from './forgot-password.component.ionic';
import { ForgotPasswordComponentCore } from './forgot-password.component.core';
var ForgotPasswordComponent = /** @class */ (function () {
    function ForgotPasswordComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    ForgotPasswordComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    ForgotPasswordComponent.prototype.ngOnDestroy = function () { };
    ForgotPasswordComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic' ?
            new ComponentMount(ForgotPasswordComponentIonic, { authState: this.authState }) :
            new ComponentMount(ForgotPasswordComponentCore, { authState: this.authState });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    ForgotPasswordComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-forgot-password',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    ForgotPasswordComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    ForgotPasswordComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return ForgotPasswordComponent;
}());
export { ForgotPasswordComponent };
