// aws-amplify-angular/dist/src/components/storage/s3-image-component/s3-image.factory.js
import { Component, Input, ViewChild, ComponentFactoryResolver, Output, EventEmitter } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { S3ImageComponentIonic } from './s3-image.component.ionic';
import { S3ImageComponentCore } from './s3-image.component.core';
var S3ImageComponent = /** @class */ (function () {
    function S3ImageComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.selected = new EventEmitter();
    }
    S3ImageComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    S3ImageComponent.prototype.ngOnDestroy = function () { };
    S3ImageComponent.prototype.loadComponent = function () {
        var _this = this;
        var imageComponent = this.framework && this.framework.toLowerCase() === 'ionic' ? new ComponentMount(S3ImageComponentIonic, { path: this.path }) : new ComponentMount(S3ImageComponentCore, { path: this.path });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(imageComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = imageComponent.data;
        componentRef.instance.selected.subscribe(function (e) {
            _this.selected.emit(e);
        });
    };
    S3ImageComponent.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-s3-image',
                    template: "\n              <div>\n                <ng-template component-host></ng-template>\n              </div>\n            "
                },] },
    ];
    /** @nocollapse */
    S3ImageComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    S3ImageComponent.propDecorators = {
        "framework": [{ type: Input },],
        "path": [{ type: Input },],
        "selected": [{ type: Output },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
    return S3ImageComponent;
}());
export { S3ImageComponent };
