// aws-amplify-angular/dist/src/components/storage/s3-image-component/s3-image.component.core.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n  <img\n    class=\"amplify-image\"\n    (click)=\"onImageClicked()\"\n    src=\"{{url}}\"\n  />\n";
var S3ImageComponentCore = /** @class */ (function () {
    function S3ImageComponentCore(amplifyService) {
        this.selected = new EventEmitter();
        this.amplifyService = amplifyService;
    }
    S3ImageComponentCore.prototype.onImageClicked = function () {
        this.selected.emit(this.url);
    };
    Object.defineProperty(S3ImageComponentCore.prototype, "data", {
        set: function (data) {
            var _this = this;
            if (!data.imagePath) {
                return;
            }
            this.amplifyService.storage()
                .get(data.imagePath)
                .then(function (url) { return _this.url = url; })
                .catch(function (err) { return console.error(err); });
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(S3ImageComponentCore.prototype, "path", {
        set: function (imagePath) {
            var _this = this;
            if (!imagePath) {
                return;
            }
            this.amplifyService.storage()
                .get(imagePath)
                .then(function (url) { return _this.url = url; })
                .catch(function (err) { return console.error(err); });
        },
        enumerable: true,
        configurable: true
    });
    S3ImageComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-s3-image-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    S3ImageComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    S3ImageComponentCore.propDecorators = {
        "selected": [{ type: Output },],
        "data": [{ type: Input },],
        "path": [{ type: Input },],
    };
    return S3ImageComponentCore;
}());
export { S3ImageComponentCore };
